chart/openfaas/templates/gateway-dep.yaml

```diff
@@ -78,8 +83,12 @@ spec:
             - --spider
             - http://localhost:8080/healthz
           {{- end }}
-          timeoutSeconds: 5
         readinessProbe:
```

Comment:
What would happen for people who have a values.yaml file that they maintain separately from the main chart?

Do we need some kind of defaulting or fallback? cc @welteki

Reply:
What do you mean? If someone provides a value at `gateway.livenessProbe.X`, it shouldn't do anything in the current chart, so I have no idea why they'd put a value there.

If they're altering the templates or something provided by this chart, that's "here be dragons" territory, where support wouldn't really exist.

Reply:
What do I mean?

What would happen here?

```
       livenessProbe:
           initialDelaySeconds: {{ .Values.gateway.livenessProbe.initialDelaySeconds }}
```

If there was someone who had their own values file and didn't have this new section you've added, called `gateway.livenessProbe`? I suspect "0" or "" would be inserted, or maybe it would error.

Whenever we add fields, we try to make sure it won't break for any users.

So I'd suggest to get this merged, we need some form of or and default value that is the same as what you get prior to this PR.

Reply:
Oh, I see what you mean. If someone updates the chart but doesn't update their values file, too. Makes sense.

Reply:
I went with the `dig | default` pattern that I use in many charts and templates. Clean and easy way to template values, using what is effectively optional chaining. It seems that the linter doesn't like it, though?

Reply:
If `.Values.favorites` doesn't exist, the template will get a nil pointer exception when trying to access `.drink`

Reply:
Here's an example: https://go.dev/play/p/tSsm8RoFqIN

Reply:
Helm uses sprig v3, which is stated in their docs.

https://github.com/helm/helm/blob/main/pkg/engine/funcs.go

They seem to override a function or two from sprig with a helm variant, but I think the contract of those functions is effectively the same.